tests/projects/test_local_pipeline.py

```diff
@@ -71,7 +71,8 @@ def test_import_artifacts(self, rundb_mock):
             assert artifact.metadata.key in expected_body_map
             assert expected_body_map[artifact.metadata.key] == artifact._get_file_body()
 
-            some_artifact = project.get_artifact(artifact.metadata.key)
+            db_key = artifact.db_key or artifact.metadata.key
+            some_artifact = project.get_artifact(db_key)
```

Comment:
why was this needed? I would expect that
```
project.get_artifact(artifact.metadata.key)
``` 
will still work

Reply:
The way we get/list artifacts with `key` is by filtering the db `key` column value according to the given key.
If an artifact is saved with a custom `db_key`, that will be the value in the `key` db column.
To query an artifact according to the value in `metadata.key` would mean to decode the blob and then filter - which is exactly what we tried to avoid in the refactor.

In this PR, the change is to use the value in `spec.db_key` as the key when you import/register an artifact, if it already exists.
the artifact used in the test has a `spec.db_key = x` value.